icu4c/source/common/localebuilder.cpp

```diff
@@ -228,9 +246,8 @@ LocaleBuilder& LocaleBuilder::setExtension(char key, StringPiece value)
         return *this;
     }
     if (extensions_ == nullptr) {
-        extensions_ = new Locale();
```

Comment:
I think the fix could be simplfy to just
extensions_ = Locale::getRoot().clone();

It does not really need to know the default locale but just try to get a locale. It is using that locale as a way to copy the extension only.

Reply:
Thanks Frank, that is a good idea. I'll update the PR to use that approach instead. Thank you!

Reply:
I pushed a new commit to use your suggestion instead.
Thanks!